agent/engine/docker_image_manager.go

```diff
@@ -267,6 +270,12 @@ func (imageManager *dockerImageManager) removeExistingImageNameOfDifferentID(con
 }
 
 func (imageManager *dockerImageManager) StartImageCleanupProcess(ctx context.Context) {
+	// If the image pull behavior is prefer cached, don't clean up the image,
+	// because the cached image is needed.
+	if imageManager.imagePullBehavior == config.ImagePullPreferCachedBehavior {
+		seelog.Info("Pull behavior is set to always use cache. Disabling cleanup")
```

Comment:
Can we point this out in the readme file?

Reply:
I think it might be valuable to add a link to the docs in `README`.

Reply:
I will ask @joelbrandenburg if it's ok to have a link in readme. I will attach the link in readme if he has the link ready before I merge, otherwise I will attach it some time after I merge.